Pid32 h automate novacustom tests v1

philipanda opened the pull request:
Draft

philipanda committed:
lib/options/: Added keywords for reading UEFI Entries

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compat../custom-network-boot-entries: Changed to use new keyword

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/uefi-setup-menu: Added RETURN KW I forgot

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/uefi-setup-menu.robot: Fix whitespace

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compat/efibootmgr.robot: Added efibootmgr test suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/uefi-setup-menu.robot: Move Cbmem keywords here from a test suite

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-perf/boot-time-measure.robot: Moved keywords to lib/options/

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/uefi-setup-menu.robot: Moved boot time measures to keywords

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/uefi-setup-menu.robot: Fix inconsistent naming

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-performance/boot-time-measure.robot: Make use of keywords

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-performance/boot-time-measure.robot: Fix variable name mistake

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu: Copy cbmem boot time keywords

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu: Add boot time measurements keywords

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-perf/boot-time-measure: Remove hardcoded resource

A config file should determine which /lib/option resource file to use

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu: Changed the kwds to work on ssh

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

mkopec commented on dasharo-compatibility/efibootmgr.robot:
```diff
@@ -0,0 +1,86 @@
+
+
+*** Test Cases ***
+EBM001.001 Network Boot enable, disable
```
Typically we have separate tests for enable and disable

philipandag replied:
I figured it may not make much sense to test enabling something if there is a chance it is already enabled because we may think that the enabling feature worked when in reality it just didn't change anything. That may cause tests to work incorrectly when done seperately or in wrong order.

Should I in this case test if NetworkBoot is disabled before enabling it and say the test failed if starting conditions were not right?

Or would it make more sense to trust that the disabling feature works and use it to make sure NetworkBoot is disabled? But then why not test if it actually worked in one go? 

Or should I just assume the starting conditions are right? I recon that's the case in other tests.

mkopec replied:
For preparing platform state before tests, you can add keywords to configure the platform correctly to the  `Suite Setup` keyword ([here](https://github.com/Dasharo/open-source-firmware-validation/pull/340/files#diff-0cf6e1c840f878221aa7bed0f49ae2c8363bf2ce1b4ccb2f61b53de8383a2cafR22). That's what we did in https://github.com/Dasharo/open-source-firmware-validation/blob/f5bead6c91692e1759ed62d2de1ce913db442975/dasharo-security/me-neuter.robot#L27 for example.

> I figured it may not make much sense to test enabling something if there is a chance it is already enabled because we may think that the enabling feature worked when in reality it just didn't change anything. 

In this case either the enable or disable test will fail, and I think it makes sense to have these tested separately, so that we can see on first glance of the results which state is not working correctly. I think that was the idea when the other tests were written, anyway.

What you're proposing also makes sense, but in this case I think I'd follow the established convention. Maybe @macpijan can chime in :)

philipandag replied:
> in this case I think I'd follow the established convention.

Sure, I am not trying to argue with your conventions.  I can't easily understand them yet as I am still very much unfamiliar with the codebase and the robot framework overall. Thanks for explaining that to me.

philipanda committed:
lib/options/uefi-setup-menu.robot: Remove non-verbose variants

There were no such variants originally. Probably I shouldn't have even added
them in the first place.

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Adjust timeouts

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-performance/boot-time-measure.robot: Adjust to new kwd names

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Changed to use more generic kwd

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-performance/boot-time-measure.robot: Remove debugging leftover

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Adjust timeout

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Adjust timeouts

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
multiple files: pre-commit formatting fix

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Adjust rtcwake time

I may have gone too far with 60 seconds. If systemd decides to be
sluggish even 60 seconds won't be enough but normally it will greatly
impact the length of the test.

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compatibility/efibootmgr.robot: Fix booting to test entry

Also added numerous little changes like seperating some functionalities
to keywords.

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compatibility/efibootmgr.robot: Move keywords section

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compatibility/device-detection.robot: Test usb stack on/off

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compatibility/device-detection: Test device detection

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
dasharo-compatibility/device-detection: Test net Controller detection

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
lib/options/dcu.robot: Add keywords for warmboot & suspension

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

mkopec commented:
@philipandag should this still be a draft?

philipandag commented:
> @philipandag should this still be a draft?
@mkopec 
Not all tests are working, so I suppose not. I have explained it on Jira.

macpijan commented:
@BeataZdunczyk @mkopec are we going to use it for the upcoming release cycle? What is the desired state of this PR?
